src/context/video-context.js

```diff
@@ -0,0 +1,240 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useRef,
+} from "react";
+import { useData } from "./database-context";
+const PlayListContext = createContext();
+
+export const PlayListProvider = ({ children }) => {
+  const { videoList } = useData();
+  const videos = Object.assign(videoList);
+  useEffect(() => {
+    dispatch({ type: "SETVIDEOLIST", value: videos });
+  }, [videos]);
+  const managePlaylistReducer = (state, action) => {
+    switch (action.type) {
+      case "INITIAL_STATE":
+        return {
+          ...state,
+          playVideo: null,
+          showPlayListOptions: false,
+          navPlaylistSelection: null,
+          showNav: false,
+          searchText: "",
+        };
+      case "SHOW_NAV":
+        return { ...state, showNav: !state.showNav };
+      case "CLOSE_NAV":
+        return { ...state, showNav: false };
+      case "SETVIDEOLIST":
+        return { ...state, videolist: action.value };
+      case "PLAYVIDEO":
+        return { ...state, playVideo: action.value };
+      case "SEARCH_ACTION":
+        return { ...state, searchText: action.value };
+      case "ADD_VIDEO_TO_HISTORY":
+        return {
+          ...state,
+          history: [
+            action.value,
+            ...state.history.filter((video) => video.id !== action.value.id),
+          ],
+        };
+      case "REMOVE_VIDEO_FROM_HISTORY":
+        return {
+          ...state,
+          history: state.history.filter(
+            (video) => video.id !== action.value.id
+          ),
+        };
+      case "REMOVE_VIDEO_FROM_WATCH_LATER":
+        return {
+          ...state,
+          watchlater: state.watchlater.filter(
+            (video) => video.id !== action.value.id
+          ),
+        };
+      case "PLAYLIST_OPTIONS":
+        return { ...state, showPlayListOptions: !state.showPlayListOptions };
+      case "ADD_TO_WATCH_LATER":
+        return {
+          ...state,
+          watchlater: action.value.addOrRemove
+            ? [...state.watchlater, action.value.video]
+            : state.watchlater.filter(
+                (video) => video.id !== action.value.video.id
+              ),
+          videolist: state.videolist.map((video) =>
+            video.id === action.value.video.id
+              ? { ...video, watchlater: !video.watchlater }
+              : video
+          ),
+          playVideo: {
+            ...state.playVideo,
+            watchlater: !state.playVideo?.watchlater,
+          },
+        };
```

Comment:
this is just a suggestion, avoid writing ternary operator that are this big. it's very hard to read and wrap my head around

Reply:
Yeah I tried to make more code clean